ui/apps/platform/src/hooks/useFetchNamespaceDeployments.ts

```diff
@@ -50,10 +50,10 @@ function useFetchNamespaceDeployments(selectedNamespaceIds: string[]) {
 
     useEffect(() => {
         if (!data || !data.results) {
-            return;
+            setAvailableNamespaces([]);
+        } else {
+            setAvailableNamespaces(data.results);
         }
```

Comment:
can we add `return () => setAvailableNamespaces([])` at the end of the useEffect to clean up state on unmount?